hls4ml/optimization/dsp_aware_pruning/keras/__init__.py

```diff
@@ -121,7 +117,7 @@ def optimize_model(
     model.compile(optimizer, loss_fn, metrics=[validation_metric])
     baseline_performance = model.evaluate(validation_dataset, verbose=0, return_dict=False)[-1]
     if verbose:
-        print(f'Baseline performance on validation set: {baseline_performance}')
+        tf.print(f'Baseline performance on validation set: {baseline_performance}')
```

Comment:
Will this actually print the statement, or will it "maybe print" depending on some external case?

Reply:
In tf execution environment (e.g., tf.function with jit), it is supposed to print, and it worked for me in other cases. Did not have the setup locally to run this part specifically, thus not tested here.

Reply:
The NumPy flag is only required for printing the loss tensors during training. Everything else should work fine with normal Python prints. However, there are no tests that run the full dsp-optimization flow and train a model with sparsity, so we need to make sure that the behaviour still stays the same and that the loss tensors are printed.